src/fu-engine.c

```diff
@@ -4490,9 +4490,15 @@ fu_engine_get_details(FuEngine *self, FuEngineRequest *request, gint fd, GError
 	for (guint i = 0; i < components->len; i++) {
 		XbNode *component = g_ptr_array_index(components, i);
 		FuDevice *dev;
-		dev = fu_engine_get_result_from_component(self, request, component, error);
-		if (dev == NULL)
-			return NULL;
+		error_local = NULL;
+
+		dev = fu_engine_get_result_from_component(self, request, component, &error_local);
```

Comment:
no, I think we should do this one layer further in, i.e. if `fu_engine_check_requirements()` in `fu_engine_get_result_from_component` fails then use an error_local and do `fu_device_inhibit()` with the error message. That way we still return a device, and the user can see why the update can't be installed.

Reply:
I've noticed that sometimes we want `inhibit` a device and add it some error message while it's not `updatable` at that moment. In this case, I've removed the `flag` check in `libfwupdplugin`.

Reply:
I've fixed it up (and added some tests) -- can you check this new version still works for you?

Reply:
LGTM, it works nicely.